Advertisement/iOS/README.md

```diff
@@ -0,0 +1,211 @@
+and navigate to Advertisement -> iOS
+Run `$ pod update` in order to set the workspace.
+
+Open `ZappAdvertisementPluginExample-iOS.xcworkspace` with Xcode 10.0.
+
+## Implementation
+The Zapp advertisement plugin API enables developers to integrate different advertisement providers to the the Zapp Platform.
+
+Advertisement Plugin is using the MVP pattern, so your plugin needs to implement the interface for the presenter (ZPAdPresenterProtocol) and it will hold a reference to the view (ZPAdViewProtocol) whose implementation is a real View used in Applicaster SDK.
+
+Also your plugin will need to implement ZPAdPluginProtocol, the main entry point that Applicaster SDK will use and it will be the constructor of the presenter's instances, as we will need to have one presenter per banner that we want to show.
+
+In order to access the `ZPAdPluginProtocol` and the `ZPAdPresenterProtocol`, you will need to import `ZappPlugins` framework:
+``` swift
+import ZappPlugins
+```
+
+### ZPAdPluginProtocol
+
+``` swift
+public class MyAdvertisementPlugin: NSObject, ZPAdPluginProtocol {...}
+```
+
+First of all we need to implement a variable and 2 constructors that are common to any plugin in our SDK:
+
+``` swift
+	//Dictionary containing the Plugin settings as defined in the plugin manifest
+	public let configurationJSON: NSDictionary?
+
+    required override public init() {
+        self.configurationJSON = nil
+        super.init()
+    }
+
+    required public init(configurationJSON: NSDictionary?){
+        self.configurationJSON = configurationJSON
+        super.init()
+    }
+```
+
+
+###### NOTE: Using the Zapp Plugin Configuration JSON
+
+```
+When creating a plugin in Zapp we can create custom configuration fields. This enable the Zapp user to fill relevant details for the specific plugin. More details can found in the [Zapp Plugin Manifest Format](http://developer-zapp.applicaster.com/zappifest/plugins-manifest-format.html).
+You can use that on the plugin level like this:
+```
+
+``` swift
+    guard let customParam = configurationJSON?["customParam"] as? String else {
+        print("Failed to create customParam from the plugin configuration JSON.")
+        return nil
+    }
+```
+
+After we will need to implement some methods specific for advertisement plugins.
+The first will be the creation of our presenter:
+
+``` swift
+    public func createAdPresenter(adView:ZPAdViewProtocol, parentVC:UIViewController) -> ZPAdPresenterProtocol {
+        return MyAdvertisementPresenter(adView: adView, parentVC: parentVC)
+    }
+```
+
+Here we are passing as parameters the view protocol that Applicaster SDK will implement and that our presenter is going to use to call some methods, and the parent view controller that we will need to use to present the interstitial or in case we need access to some of their parameters.
+
+After we need to implement a method that is used by our analytics system to know the name of the plugin
```

Comment:
@PRuedaApplicaster can you explain me a little bit about this? I'm not familiar with that.

Reply:
Here are the docs to know more about it:
https://docs.google.com/spreadsheets/d/1dn23ciC4Q14DQPXhIv4Kbv9woPhO_tTCayvWXVVRS-k/edit?usp=sharing
(in the second tab you can see we need some way to get the ad provider name)